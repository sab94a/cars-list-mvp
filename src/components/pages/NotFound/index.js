//@flow

import React from 'react';
import Link from 'components/lib/Link';
import Logo from 'components/lib/Logo';
import styles from './index.module.scss';

const NotFound = () => (
    <div className={ styles.root }>
        <div className={ styles.wrapper }>
            <Logo />
            <h1 className={ styles.title }>404 - Not Found</h1>
            <p className={ styles.note }>Sorry, the page you are looking for does not exist.</p>
            <p className={ styles.note }>You can always go back to the <Link href="/">homepage.</Link></p>
        </div>
    </div>
)

export default NotFound;
