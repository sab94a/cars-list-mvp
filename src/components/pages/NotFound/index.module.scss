@import 'components/lib/variables.scss';

.root {
    display: flex;
    width: 100%;
    min-height: 100%;
    align-items: center;
    justify-content: center;
}

.wrapper {
    margin: auto;
    text-align: center;
}

.title {
    margin: 0;
    padding: $offset-lg 0;
    font-size: $font-size-lg; 
}

.note {
    margin: 0;
    font-size: $font-size-md;
    padding: $offset-sm 0;

    a {
        text-decoration: none;
    }
}
