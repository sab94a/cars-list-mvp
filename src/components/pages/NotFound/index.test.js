import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import NotFound from './'

describe('<NotFound />', () => {
    it('Should render correctly', () => {
        const component = mount(<NotFound />)

        expect(toJson(component)).toMatchSnapshot();
    });
});
