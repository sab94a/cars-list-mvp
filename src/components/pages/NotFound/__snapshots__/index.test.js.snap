// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> Should render correctly 1`] = `
<NotFound>
  <div
    className="root"
  >
    <div
      className="wrapper"
    >
      <Logo>
        <img
          alt="Logo"
          className="root"
          src="logo.png"
        />
      </Logo>
      <h1
        className="title"
      >
        404 - Not Found
      </h1>
      <p
        className="note"
      >
        Sorry, the page you are looking for does not exist.
      </p>
      <p
        className="note"
      >
        You can always go back to the 
        <Link
          href="/"
        >
          <a
            className="root"
            href="/"
          >
            homepage.
          </a>
        </Link>
      </p>
    </div>
  </div>
</NotFound>
`;
